src/Address.php

```diff
@@ -7,47 +7,55 @@
 use function BitWasp\Bech32\decode;
 use function BitWasp\Bech32\encode;
 use InvalidArgumentException;
+use Throwable;
 
 class Address
 {
-    const HRP = 'erd';
+    const DEFAULT_HRP = 'erd';
     const CONTRACT_HEX_PUBKEY_PREFIX = '0000000000000000';
     const BECH32_ADDRESS_LENGTH = 62;
+    const PUBKEY_LENGTH = 32;
 
     private function __construct(
         private string $valueHex,
+        public readonly string $hrp = self::DEFAULT_HRP
     ) {
     }
 
-    public static function fromHex(string $value): Address
+    public static function newFromHex(string $value, string $hrp = self::DEFAULT_HRP): Address
     {
+        if (! self::isValidHex($value)) {
+            throw new InvalidArgumentException('invalid hex value');
+        }
+
         return new Address(
-            $value ?: throw new InvalidArgumentException('hex value is required')
+            $value ?: throw new InvalidArgumentException('hex value is required'),
+            $hrp
         );
     }
 
-    public static function fromBech32(string $address): Address
+    public static function newFromBech32(string $address): Address
     {
         if (strlen($address) !== self::BECH32_ADDRESS_LENGTH) {
             throw new Exception('invalid address length');
         }
 
-        $decoded = decode(strtolower($address))[1];
+        [$hrp, $decoded] = decode(strtolower($address));
         $res = convertBits($decoded, count($decoded), 5, 8, false);
         $pieces = array_map(fn ($bits) => dechex($bits), $res);
         $hex = array_reduce($pieces, fn ($carry, $hex) => $carry . str_pad($hex, 2, "0", STR_PAD_LEFT));
 
-        return new Address($hex);
+        return new Address($hex, $hrp);
     }
 
-    public static function fromBase64(string $value): Address
+    public static function newFromBase64(string $value, string $hrp = self::DEFAULT_HRP): Address
```

Comment:
Good point, maybe we should add this in specs. Useful when handling (at a lower level) topics of some events etc.

Reply:
but.. why do we need to rename the function name, if the second added param has a default? It would still be BC... right?

Reply:
@razvanphp The rename is part of aligning with the new sdk specs. Note that we did not remove the old ones like `fromBase64` but only deprecated them with a reference to the new functions